subsys/net/ip/net_core.c

```diff
@@ -65,12 +66,47 @@ void net_analyze_stack(const char *name, const char *stack, size_t size)
 }
 #endif /* CONFIG_INIT_STACKS */
 
+#if defined(CONFIG_NET_SOCKETS_RAW)
+static inline enum net_verdict process_raw_data(struct net_pkt *pkt)
+{
+	struct net_context *ctx;
+
+	ctx = net_context_find(AF_PACKET, SOCK_RAW, ETH_P_ALL);
+	if (!ctx) {
+		return NET_DROP;
+	}
+
+	/* Currently we are skipping L2 layer verification and not
+	 * removing L2 header from packet.
+	 * TODO :
+	 * 1) Pass it through L2 layer, so that L2 will verifies
+	 * that packet is intended to us or not and sets src and dst lladdr.
+	 * And L2 should not pullof L2 header when combination of socket
+	 * like this AF_PACKET, SOCK_RAW and ETH_P_ALL proto.
+	 * 2) Socket combination of AF_INET, SOCK_RAW, IPPROTO_RAW
+	 * packet has to go through L2 and L2 verfies it's header and removes
+	 * header. Only packet with L3 header will be given to socket.
+	 * 3) If user opens raw and non raw socket together, based on raw
+	 * socket combination packet has to be feed to raw socket and only
+	 * data part to be feed to non raw socket.
+	 */
+
+	net_pkt_set_family(pkt, AF_PACKET);
+
+	return net_conn_input(ETH_P_ALL, pkt);
+}
+#endif
+
 static inline enum net_verdict process_data(struct net_pkt *pkt,
 					    bool is_loopback)
 {
 	int ret;
 	bool locally_routed = false;
 
+#if defined(CONFIG_NET_SOCKETS_RAW)
+	return process_raw_data(pkt);
```

Comment:
I still don't get this, so we can't have SOCK_STREAM, SOCK_DGRAM (and thus ipv4, ipv6, udp, tcp....) at same time as SOCK_RAW?

Reply:
Yes, We can not open raw and non-raw socket at the moment. I will be working on it later.